app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,154 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export_id)
+    ActiveRecord::Base.connection_pool.with_connection do
+      csv_export = CsvExport.find(csv_export_id)
+      create_export_folder
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # clean up downloaded files older than 12 hours or any jobs stuck for over 1 month
+    # TODO delete files with no record associations
+    end_date = Time.now - Rails.application.config.samson.export_job.downloaded_age
+    timeout_date = Time.now - Rails.application.config.samson.export_job.max_age
+    @csv_exports = CsvExport.where("(status = 'downloaded' AND updated_at <= :end_date) OR updated_at <= :timeout_date",
+      {end_date: end_date, timeout_date: timeout_date})
+    if @csv_exports
+      @csv_exports.each do |csv_downloaded|
+        filename = csv_downloaded.full_filename
+        File.delete(filename) unless !File.exist?(filename)
+        csv_downloaded.delete
+        Rails.logger.info("Downloaded file #{csv_downloaded.filename} deleted")
+      end
+    end
+  end
+  
+  def export_task(csv_export)
+    if csv_export.content == "deploys"
+      csv_export.started!
+      deploy_csv_export(csv_export)
+    else
+      csv_export.failed!
+    end
+
+    notify_of_creation(csv_export)
+  rescue
+    #delete_file(csv_export.full_filename)
```

Comment:
I'd expect errors to be in ruby land and not halfway through generating the file and we don't want to ship a half-built file ... so better get rid of it ...